File: Data Structures & Algorithms/pow-x-n/submission-0.java
class Solution {
    public double myPow(double x, int n) {
        double res = 1;
        if(n == 0)
        {
            return res;
        }
        while(n != 0)
        {
            if(n > 0)
            {
                res = res * x;
                n--;
            }
            else
            {
                double val = 1;
                val = val / x;
                res = res * val;
                n++;
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        res = []
        prefix = []
        postfix = []
        for i in range(len(nums)):
            res.append(0)
            if i == 0:
                prefix.append(nums[i])
            else:
                prefix.append(1)
            if i == len(nums) - 1:
                postfix.append(nums[i])
            else:
                postfix.append(1)
        
        j = len(nums) - 2
        for i in range(1,len(nums)):
            prefix[i] = prefix[i-1] * nums[i]
            postfix[j] = postfix[j+1] * nums[j]
            j -= 1
        print(prefix,postfix)
        for i in range(len(nums)):
            if i == 0:
                res[i] = postfix[i + 1]
            elif i == len(nums) - 1:
                res[i] = prefix[i-1]
            else:
                res[i] = prefix[i-1] * postfix[i + 1]
        return res


